src/Avalonia.Controls/Primitives/PopupRoot.cs

```diff
@@ -154,7 +154,11 @@ void IPopupHost.ConfigurePosition(PopupPositionRequest request)
             UpdatePosition();
         }
 
-        public void SetChild(Control? control) => Content = control;
+        public void SetChild(Control? control)
+        {
+            PlatformImpl?.SetTopmost(Topmost);
```

Comment:
Feels a bit wrong to do anything unrelated to `SetChild` in this method. Also, wouldn't it break if Topmost changed in runtime?

Reply:
Yeah, I wasn't sure about that one. The TopMost property is set in the line above in [Popup.cs](https://github.com/AvaloniaUI/Avalonia/blob/07f3ad23e49da9ced46b7a68392e78a150622c35/src/Avalonia.Controls/Primitives/Popup.cs#L431), so I thought setting the TopMost here would be least intrusive. Changing and affecting the property at runtime would need to handle it in `OnPropertyChanged` of the PopupRoot? Would that approach be better?

Reply:
Check the last commit: Seems to work and doesn't feel "wrong" anymore? 😅

Reply:
Popup already has OnPropertyChanged for Topmost:
https://github.com/AvaloniaUI/Avalonia/blob/07f3ad23e49da9ced46b7a68392e78a150622c35/src/Avalonia.Controls/Primitives/Popup.cs#L652

Reply:
As for PopupRoot, probably mirror behavior of WindowShadowHint property?
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Primitives/PopupRoot.cs#L63
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Primitives/PopupRoot.cs#L224-L227

Reply:
@StefanKoell just noticed newer commit. Yes, that's much better.

Reply:
Got it! Should now work pretty much the same as with WindowShadowHint.